src/Symfony/Component/Process/PhpSubprocess.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Process;
+
+use Symfony\Component\Process\Exception\LogicException;
+use Symfony\Component\Process\Exception\RuntimeException;
+
+/**
+ * PhpSubprocess runs a PHP command as a subprocess while keeping the original php.ini settings.
+ *
+ * For this, it generates a temporary php.ini file taking over all the current settings and disables
+ * loading additional .ini files. Basically, your command gets prefixed using "php -n -c /tmp/temp.ini".
+ *
+ * Given your php.ini contains "memory_limit=-1" and you have a "MemoryTest.php" with the following content:
+ *
+ * <?php var_dump(ini_get('memory_limit'));
+ *
+ * These are the differences between the regular Process and PhpSubprocess classes:
+ *
+ *     $p = new Process(['php', '-d', 'memory_limit=256M', 'MemoryTest.php']);
+ *     $p->run();
+ *     print $p->getOutput()."\n";
+ *
+ * This will output "string(2) "-1", because the process is started with the default php.ini settings.
+ *
+ *     $p = new PhpSubprocess(['MemoryTest.php'], null, null, 60, ['php', '-d', 'memory_limit=256M']);
+ *     $p->run();
+ *     print $p->getOutput()."\n";
+ *
+ * This will output "string(4) "256M"", because the process is started with the temporarily created php.ini settings.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ * @author Partially copied and heavily inspired from composer/xdebug-handler by John Stevenson <john-stevenson@blueyonder.co.uk>
+ */
+class PhpSubprocess extends Process
+{
+    /**
+     * @param array       $command The command to run and its arguments listed as separate entries. They will automatically
+     *                             get prefixed with the PHP binary
+     * @param string|null $cwd     The working directory or null to use the working dir of the current PHP process
+     * @param array|null  $env     The environment variables or null to use the same environment as the current PHP process
+     * @param int         $timeout The timeout in seconds
+     * @param array|null  $php     Path to the PHP binary to use with any additional arguments
+     */
+    public function __construct(array $command, string $cwd = null, array $env = null, int $timeout = 60, array $php = null)
+    {
+        if (null === $php) {
+            $executableFinder = new PhpExecutableFinder();
+            $php = $executableFinder->find(false);
+            $php = false === $php ? null : array_merge([$php], $executableFinder->findArguments());
+        }
+
+        $tmpIni = $this->writeTmpIni($this->getAllIniFiles(), sys_get_temp_dir());
+
+        if ($tmpIni) {
+            $php = array_merge($php, ['-n', '-c', $tmpIni]);
```

Comment:
if `ExecutableFinder` returns `false`, `$php` will still be `null` at this point.